### PR TITLE
Simplify `conj` and `real` for `::Real`

### LINKED ISSUE #205: conj(@syms x::Real) ≠ x
I was trying some things out with ModelingToolkit and I ran across this conjugate of Reals, when using the dot product with `Vector{Sym{Real}}`, that is not being simplified

```julia
using SymbolicUtils
@syms x::Real
simplify(conj(x))
```

It would be nice if this simplifies. Not sure how to help with that, the code generation in `methods.jl` is not easy for me.

### PR DESCRIPTION
**Pull request by Suavesito-Olimpiada**
Fix #205

### CODE REVIEW

**Comment by Suavesito-Olimpiada** [comment]
Fix an additional case where
```julia
@syms x::Real
simplify(real(x)) != x
```
adding `_isreal` to the `utils.jl`.

**Comment by shashi** [comment]
Nice! Since we have also been simplifying "on construction", could you add a method to conj for Symbolic{<:Real} and Symbolic{<:Number}?

**Review by shashi — APPROVED** [review]

**Comment by Suavesito-Olimpiada** [comment]
> Nice! Since we have also been simplifying "on construction", could you add a method to conj for Symbolic{<:Real} and Symbolic{<:Number}?

Where should I add these methods? I thought about it but didn't find the correct place. :b

**Comment by Suavesito-Olimpiada** [comment]
Maybe I see, is `methods.jl` the correct file to place this?

**Comment by Suavesito-Olimpiada** [comment]
I think it should be fine by now. :)

File: src/methods.jl [hunk 1]
```diff
@@ -111,8 +111,15 @@ for f in [identity, one, zero, *, +, -]
     @eval promote_symtype(::$(typeof(f)), T::Type{<:Number}) = T
 end
 
+# All of these need to be `Sym` to overwrite the function definitions of
+# the L78, that are made with the type `Sym <: Symbolic`
+#   @number_methods(Sym, term(f, a), term(f, a, b), skipbasics)
 promote_symtype(::typeof(Base.real), T::Type{<:Number}) = Real
-Base.real(s::Symbolic{<:Number}) = islike(s, Real) ? s : term(real, s)
+Base.real(s::Sym{<:Real}) = s
```
Comment (shashi) [inline src/methods.jl:118]: why is this only `Sym` and not `Symbolic`?

Comment (shashi) [reply src/methods.jl:118]: Oh I just read your comment. Well this seems like it should not be defined in L78, but should be defined outside of the loop there (but could be within `@number_methods` macro). So we should probably remove `real`, `conj` and `imag` from the list of one-argument functions.

File: test/rulesets.jl [hunk 2]
```diff
@@ -18,6 +18,9 @@ end
 
 @testset "Numeric" begin
     @syms a::Integer b c d x::Real y::Number
+    @eqtest simplify(conj(x)) == x
+    @eqtest simplify(real(x)) == x
+    @eqtest simplify(imag(x)) == 0
```
Comment (shashi) [inline test/rulesets.jl:23]: You will need to update the tests because `conj(x)` may just return `x`. Use `Term{Real}(conj, [x])` instead.

Comment (Suavesito-Olimpiada) [reply test/rulesets.jl:23]: Ok, ok.

File: src/methods.jl (hunk 1 above)
Comment (Suavesito-Olimpiada) [reply src/methods.jl:118]: I'll move them and update the PR.

Comment (Suavesito-Olimpiada) [reply src/methods.jl:118]: Should I force push or add a commit?

Comment (shashi) [reply src/methods.jl:118]: New commit is fine. Either is fine.

**Pull request merged**